src/js/App/GlobalFilter/TagsModal.tsx

```diff
@@ -7,126 +7,132 @@ import debounce from 'lodash/debounce';
 import flatMap from 'lodash/flatMap';
 import { useIntl } from 'react-intl';
 import messages from '../../Messages';
-import { GlobalFilterTag, ReduxState, SID } from '../../redux/store';
+import { CommonSelectedTag, CommonTag, GlobalFilterTag, ReduxState, SID } from '../../redux/store';
 import { FlagTagsFilter } from './constants';
-import { TagPagination } from './tagsApi';
+import { TagFilterOptions, TagPagination } from './tagsApi';
+import { Action } from 'redux';
 
 export type TagsModalProps = {
   isOpen?: boolean;
   filterTagsBy: string;
   toggleModal: (isSubmit: boolean) => void;
   selectedTags?: FlagTagsFilter;
-  onApplyTags: (tags: GlobalFilterTag[], sids: SID[]) => void;
+  onApplyTags: (tags: CommonSelectedTag[], sids: CommonSelectedTag[]) => void;
 };
 
-const useMetaSelector = (key: 'tags' | 'workloads' | 'sid') =>
+export type IDMapper = (tag: CommonTag) => string;
+export type CellsMapper = (tag: CommonTag) => (string | number | boolean | undefined)[];
+export type DebounceCallback = (filters?: TagFilterOptions, pagination?: TagPagination) => Action;
+
+export const useMetaSelector = (key: 'tags' | 'workloads' | 'sid') =>
   useSelector<ReduxState, [boolean | unknown, number, number, number]>(({ globalFilter }) => {
     const selected = globalFilter[key];
     return [selected?.isLoaded, selected?.total || 0, selected?.page || 1, selected?.perPage || 10];
   }, shallowEqual);
 
-const TagsModal = ({ isOpen, filterTagsBy, onApplyTags, toggleModal, selectedTags }: TagsModalProps) => {
-  const intl = useIntl();
-  const [tagsSelected, setTagsSelected] = useState<GlobalFilterTag[]>([]);
-  const [sidsSelected, setSidsSelected] = useState<SID[]>([]);
-  const [filterBy, setFilterBy] = useState('');
-  const [filterSIDsBy, setFilterSIDsBy] = useState('');
-  const dispatch = useDispatch();
-  const [tagsLoaded, tagsCount, tagsPage, tagsPerPage] = useMetaSelector('tags');
-  const [sidLoaded, sidCount, sidPage, sidPerPage] = useMetaSelector('sid');
-  const tags = useSelector<ReduxState, GlobalFilterTag[]>(({ globalFilter: { tags } }) => tags?.items || []);
-  const sids = useSelector<ReduxState, SID[]>(({ globalFilter: { sid } }) => sid?.items || []);
-  const filterScope = useSelector<ReduxState, 'insights' | undefined>(({ globalFilter: { scope } }) => scope || undefined);
-  const debounceGetTags = useCallback(
-    debounce((search?: string) => {
-      dispatch(
-        fetchAllTags(
-          {
-            registeredWith: filterScope,
-            activeTags: selectedTags,
-            search,
-          },
-          { page: tagsPage, perPage: tagsPerPage }
-        )
+const usePagination = (loaded: boolean | unknown, perPage?: number, page?: number, count?: number) => {
+  return useMemo(() => {
+    if (loaded) {
+      return {
+        perPage: perPage,
+        page: page,
+        count: count,
+      };
+    }
+    return {};
```

Comment:
I am wondering if we maybe want to return attributes with the value 0  instead of undefined.

Reply:
I don't think so, if the pagination sees empty config it doesn't render. However if we were to sent `count: 0` it would render it and then once data comes in it would flip it to the real data counter.